hibernate-reactive-core/src/main/java/org/hibernate/reactive/mutiny/impl/MutinyStatelessSessionImpl.java

```diff
@@ -0,0 +1,96 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.mutiny.impl;
+
+import io.smallrye.mutiny.Uni;
+import org.hibernate.LockMode;
+import org.hibernate.reactive.common.ResultSetMapping;
+import org.hibernate.reactive.mutiny.Mutiny;
+import org.hibernate.reactive.session.ReactiveStatelessSession;
+
+/**
+ * Implements the {@link Mutiny.StatelessSession} API. This delegating
+ * class is needed to avoid name clashes when implementing both
```

Comment:
Is that the only reason? Couldn't use fully qualified names?

Reply:
For *method names*.

Reply:
ah, ok!